app/apollo/resolvers/channel.js

```diff
@@ -694,12 +712,30 @@ const channelResolvers = {
 
       const user = whoIs(me);
 
-      try{
-        logger.info({req_id, user, org_id, uuid }, `${queryName} validating`);
+      try {
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'uuid', uuid );
 
+        // Find version (avoid using deprecated/ignored `versions` attribute on the channel)
+        const version = await models.DeployableVersion.findOne( { uuid, org_id } );
+
+        const channel = await models.Channel.findOne( { uuid: version.channel_id, org_id } );
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - channel found: ${!!channel}`);
+
+        const identifiers = channel ? [uuid, channel.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.MANAGEVERSION, TYPES.CHANNEL, queryName, context, identifiers);
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - authorized`);
+
+        if( !version ){
+          throw new NotFoundError( context.req.t( 'Version uuid "{{version_uuid}}" not found.', { 'version_uuid': uuid } ), context );
+        }
```

Comment:
If the version is not found, then the code _cannot_ try to find the channel with `version.channel_id`.  Therefore the check for version not found must be moved earlier than the auth check.
```suggestion
        // Find version (avoid using deprecated/ignored `versions` attribute on the channel)
        const version = await models.DeployableVersion.findOne( { uuid, org_id } );
        if( !version ){
          throw new NotFoundError( context.req.t( 'Version uuid "{{version_uuid}}" not found.', { 'version_uuid': uuid } ), context );
        }

        const channel = await models.Channel.findOne( { uuid: version.channel_id, org_id } );
        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - channel found: ${!!channel}`);

        const identifiers = channel ? [uuid, channel.name] : [uuid];
        await validAuth(me, org_id, ACTIONS.MANAGEVERSION, TYPES.CHANNEL, queryName, context, identifiers);
        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - authorized`);
```

Reply:
This was also addressed locally, something is up with the diff

Reply:
That would be because this is outdated

Reply:
I think the fault was on my end -- refreshing a diff between specific commits rather than getting the overall diff for the PR fresh.
Resolved.